### PR TITLE
Refactor logging

### LINKED ISSUE #2017: Centralize EventSource and fix severity levels
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1983/files#r1072147421

### PR DESCRIPTION
**Pull request by dszmigielski**
## Why

EventSource usage is inconsistent.

Fixes #2017

## What

Make EventSource usage consistent. Centralize EventSource.

## Tests

CI/CD

## Checklist

- ~~[ ] `CHANGELOG.md` is updated.~~
- ~~[ ] Documentation is updated.~~
- ~~[ ] New features are covered by tests.~~

### CODE REVIEW

File: src/OpenTelemetry.AutoInstrumentation/Configurations/Propagator.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ namespace OpenTelemetry.AutoInstrumentation.Configurations;
 internal enum Propagator
 {
     /// <summary>
-    /// W3C Trace Context propagator.
+    /// W3C Information Context propagator.
```
Comment (RassK) [inline src/OpenTelemetry.AutoInstrumentation/Configurations/Propagator.cs:25]: error in find & replace

**Review by rajkumar-rangaraj — CHANGES_REQUESTED** [review]
When `EventSource` is shared between different project, it will throw an error during initialization but may not crash the process. 

**Error:** An instance of EventSource with Guid already exists.

Please validate and confirm we are not throwing this error. If this error is thrown event source messages will not be emitted.

**Comment by pellared** [comment]
> Extract base class for EventSource, and pass its implementation to the CustomLogger as parameter.

It seams reasonable. This is what I initially had in mind [here](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1983/files#r1072147421).

**Comment by rajkumar-rangaraj** [comment]
> > Extract base class for EventSource, and pass its implementation to the CustomLogger as parameter.
> 
> It seams reasonable. This is what I initially had in mind [here](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1983/files#r1072147421).
> 
> @rajkumar-rangaraj any preferences?

I'm fine with option 1, was worried about missing event source in the current proposal.

**Comment by Kielek** [comment]
@rajkumar-rangaraj, concept changed a bit.
With Robert we have decided to use same code twice. Just EventSourceName is different.

**Review by Kielek — APPROVED** [review]

**Review by pellared — APPROVED** [review]

**Review by RassK — APPROVED** [review]
Seems good for now but I think needs further refactoring - separating both sinks (file and event source).

**Comment by pellared** [comment]
@rajkumar-rangaraj Can you please double-check please (even after it is merged)?

**Pull request merged**